python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -278,7 +282,8 @@ def __call__(self, fn: Callable) -> AssetsDefinition:
             for key in self.required_resource_keys:
                 required_resource_keys.add(key)
             for key in self.resource_defs.keys():
-                required_resource_keys.add(key)
+                if not isinstance(self.resource_defs[key], IOManagerDefinition):
```

Comment:
got confused here by this check as well. Would be great to add a comment here to explain

Reply:
As it turns out, I'm wrong. IO manager defs _are_ available from within the body of an op.